### DISCUSSION TITLE
Save audio file from Axis API request

### DISCUSSION THREAD

**Discussion by FernandoArteaga** [AxisCommunications/discussions#59] [VAPIX]
Hi all, I am trying to make a small POC that allows me to record 5 secs of audio into a file using the API.

I am using Python and I am using the Audio API VAPIX section, specifically the "Singlepart audio data response" endpoint (`//<servername>/axis-cgi/audio/receive.cgi?httptype=singlepart`).

For the moment I can send the request to the API and I see the bites coming in, but when I try to save them into a **.wav** file I only get noise.

I think the issue can be in the way I'm saving the file, probably the settings are not well configured.

Here is the code I'm using for, thanks in advance.

```python
WAVE_OUTPUT_FILENAME = "axis.wav"
CHANNELS = 1
RATE = 128000  # 128000  --  44100
FORMAT = pyaudio.paInt16

def wave_file(filename, channels, sample_width, rate, write_frames):
    wf = wave.open(filename, 'wb')
    wf.setnchannels(channels)
    wf.setsampwidth(sample_width)
    wf.setframerate(rate)
    wf.writeframes(write_frames)
    wf.close()


if __name__ == '__main__':
    s, r = axis_client.audio_stream()  # this is the API call, nothing special inside
    if s == 'ok':
        frames = []
        start = datetime.now()
        for line in r.iter_lines():
            duration = datetime.now() - start
            frames.append(line)
            if duration.seconds > 5:
                break
        wave_file(WAVE_OUTPUT_FILENAME, 1, p.get_sample_size(FORMAT), RATE, b''.join(frames))
    else:
        print(r)
```

requirements.txt:
```text
requests==2.28.1
PyAudio==0.2.12
```

**Comment by mattias-kindborg-at-work** [comment]
Hi @FernandoArteaga!

Could you please add your imports? I'm guessing we need to import `pyaudio`, but I didn't see anything relating to `axis_client` in the requirements file?

**Reply by FernandoArteaga** [reply]
Hello @mattias-kindborg-at-work,

I've omitted the imports because I wasn't importing anything special. The `axis_client` is a custom class I made and it works as a wrapper for the HTTP request.

So `axis_client.audio_stream()` is like:

```python
class AxisClient:
    _username = 'user'
    _passwords = 'pass'
    _host = 'http://192.168.2.47/axis-cgi'

    def audio_stream(self):
       url = f'{self._host}/audio/receive.cgi'
        auth = HTTPDigestAuth(self._username, self._passwords)
        params = {
            'httptype': 'singlepart',
        }
        try:
            return 'ok', requests.get(url, auth=auth, params=params, timeout=1, stream=True)
        except Exception as e:
            return 'error', e
```

**Comment by vivekatoffice** [comment]
Look like you may need to set the correct "content-type".
